server/engine/Board.js

```diff
@@ -0,0 +1,45 @@
+const Words = require("./Words");
+const Card = require("./Card");
+const MapCard = require("./MapCard");
+
+class Board {
+  constructor() {
+    //new deck of words
+    this.deck = [...Words];
+    this.shuffle(this.deck);
+    this.deckLength = this.deck.length;
+    this.mapCard = new MapCard();
+    //drawed cards
+    this.cards = [];
+  }
+
+  generateNewRound() {
+    let selectedWords = [];
+    if (this.deckLength > 25) {
+      selectedWords = this.deck.splice(0, 25);
+      this.updateDeckLength();
+    } else {
+      this.deck = [...Words];
```

Comment:
Also, I'm not sure if we're allowed to play the game with only less than 25 words on the board (correct me if I'm wrong though).

Reply:
Oh, I was going to reset the deck when it's less than 25 words. I think I forget to select the words after reseting the deck. Thanks for pointing out.